book/src/rfcs/benchmarking.md

```diff
@@ -0,0 +1,109 @@
+| kimchi memory            | proof-systems | memory   | yes              | yes              | yes             | no               |
+| kimchi memory flamegraph | proof-systems | memory   | no               | yes              | no              | no               |
+| snarky time              | mina/snarky*  | time     | yes              | yes              | yes             | yes              |
+| snarky memory            | mina/snarky*  | memory   | yes              | yes              | yes             | no               |
+| pickles time             | mina/pickles* | time     | yes              | yes              | yes             | yes              |
+| pickles memory           | mina/pickles* | memory   | yes              | yes              | yes             | no               |
+
+\* these have to be in Mina now because pickles is in Mina, and while snarky has it's repo, to use it requires a few things still in Mina, if it were possible change that it would make these benchmarks easier and probably cheaper.
+
+### Kimchi time
+
+One of the simplest measuring running time, can be an specific case or allow more flexibility by introducing parameters like for example different circuit sizes. We already have most of the infrastructure for this with criterion. Would ideally have a dedicated computer, considering that there aren't that many PRs in proof-systems the requirements shouldn't be that high.
+
+### Kimchi time flamegraph
+
+This can be considered a variant of the previous, some more research is required but ideally can be generated from the already existing benchmarks with little to no changes.
+This may not be as interesting to have in CI, but can be useful to run locally for those working in optimizations.
+
+### Kimchi instructions
+
+This is basically counting the CPU instructions that takes to run some code, could be use as an alternative for CI given that it won't be affected by noise and can work in any computer. It would also be faster by running the code only once. The implementation will require a bit more research but one option is the crate iai, with some thinking it could be possible to have this and the time benchmarks share the same code.
+Is worth mentioning that the result are specific to the code and even the computer running the benchmark and thus not useful for comparisons with other projects.
+
+### Kimchi memory
+
+Similar to time, but with memory, the constraints are different in some ways, noise is not generally a problem and any computer cab be used, and also requires less sampling and should run faster. This is to some degree already implemented through a custom measurement for criterion, and in practice can be slower than measuring time, the instrumentation makes it run slower and the some limitations in criterion require to run the code more than necessary. With that in mind the best would be to run just a subset of these benchmark for regressions and let the others be run at discretion.
```

Comment:
```suggestion
Similar to time, but with memory, the constraints are different in some ways, noise is not generally a problem and any computer can be used, and also requires less sampling and should run faster. This is to some degree already implemented through a custom measurement for criterion, and in practice can be slower than measuring time, the instrumentation makes it run slower and the some limitations in criterion require to run the code more than necessary. With that in mind, the best would be to run just a subset of these benchmark for regressions and let the others be run at discretion.
```

Reply:
Do you already have some ideas about which benchmarks would be marked as "regressions" or not?